React/Views/RCTView.m

```diff
@@ -772,6 +783,14 @@ - (void)displayLayer:(CALayer *)layer
   // solve this, we'll need to add a container view inside the main view to
   // correctly clip the subviews.
 
+  id savedTraitCollection = nil;
```

Comment:
I started looking at the [docs from Apple](https://developer.apple.com/documentation/uikit/appearance_customization/adopting_ios_dark_mode?language=objc) on how to do this which I believe is what you referred in our discussion. I find:
```
layer.backgroundColor = [_backgroundColor resolvedColorWithTraitCollection:self.traitCollection].CGColor
```
a much more elegant solution, wouldn't that work out of the box? without the need to store and set the current trait collection. Can you also include comment to the talk just for clarity.

Reply:
Sure.   I can do that.   I think the save/set/restore `currentTraitCollection` pattern may be cleaner if there were many calls to `.CGColor`, but in this case there is only the one.

The pattern of saving/setting/restoring `currentTraitCollection` is imitating what UIKit does itself before/after calling the special methods such as `UIView` `updateLayer` and `drawRect:` as described in this [doc](https://developer.apple.com/documentation/xcode/supporting_dark_mode_in_your_interface?language=objc).   The docs of [currentTraitCollection](https://developer.apple.com/documentation/uikit/uitraitcollection/3238080-currenttraitcollection?language=objc) also describes this behavior.  Its even more explicitly described at time 22:00 of this [WWDC video](https://developer.apple.com/videos/play/wwdc2019/214/)